src/test/java/io/pravega/connectors/flink/FlinkPravegaTableITCase.java

```diff
@@ -0,0 +1,345 @@
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Integration tests for {@link FlinkPravegaTableSource}
+ */
+@Slf4j
+public class FlinkPravegaTableITCase {
+
+    // Setup utility.
+    protected static final SetupUtils SETUP_UTILS = new SetupUtils();
+
+    private static final List<Row> RESULTS = new ArrayList<>();
+
+    // Number of events to generate for each of the tests.
+    private static final int EVENT_COUNT_PER_SOURCE = 11;
+
+    //Ensure each test completes within defined interval
+    @Rule
+    public final Timeout globalTimeout = new Timeout(180, TimeUnit.SECONDS);
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        SETUP_UTILS.setEnableAuth(false);
+        SETUP_UTILS.setEnableTls(false);
+        SETUP_UTILS.startAllServices();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        SETUP_UTILS.stopAllServices();
+    }
+
+    @Test
+    public void testJsonTableSource() throws Exception {
+
+        StreamExecutionEnvironment execEnvWrite = StreamExecutionEnvironment.getExecutionEnvironment();
+        execEnvWrite.setParallelism(1);
+
+        Stream stream = Stream.of(SETUP_UTILS.getScope(), "testJsonTableSource");
+        SETUP_UTILS.createTestStream(stream.getStreamName(), 1);
+
+        String[] fieldNames = {"user", "uri", "accessTime"};
+
+        PravegaConfig pravegaConfig = SETUP_UTILS.getPravegaConfig();
+
+        // Write some data to the stream
+        DataStreamSource<Row> dataStream = execEnvWrite
+                .addSource(new TableEventSource(EVENT_COUNT_PER_SOURCE));
+
+        FlinkPravegaWriter<Row> pravegaSink = FlinkPravegaWriter.<Row>builder()
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .withSerializationSchema(new JsonRowSerializationSchema(fieldNames))
+                .withEventRouter((Row event) -> "fixedkey")
+                .build();
+
+        dataStream.addSink(pravegaSink);
+
+        Assert.assertNotNull(execEnvWrite.getExecutionPlan());
+
+        execEnvWrite.execute("PopulateRowData");
+
+        // read data from the stream using Table reader
+        TableSchema tableSchema = TableSchema.builder()
+                .field("user", Types.STRING())
+                .field("uri", Types.STRING())
+                .field("accessTime", Types.SQL_TIMESTAMP())
+                .build();
+
+        FlinkPravegaJsonTableSource source = FlinkPravegaJsonTableSource.builder()
+                                                .forStream(stream)
+                                                .withPravegaConfig(pravegaConfig)
+                                                .failOnMissingField(true)
+                                                //.withProctimeAttribute("accessTime")
```

Comment:
Why is this commented out?

Reply:
We could use either `withProctimeAttribute` or `withRowtimeAttribute` to act on the time information. I have left the commented code there just to indicate the process time attribute usage.

Reply:
Would it make sense to have a test case for each? Also, what's the expected behavior in the case the user accidentally uses both.

Reply:
supplying both `proc` and `row` time attributes will throw an exception and hence cannot be used together. Having a test case for `proc` time is not required since it is based on the wall clock time and will have no bearing on the record timestamp.
> "org.apache.flink.table.api.ValidationException: Field 'accessTime' must not be processing time and rowtime attribute at the same time."